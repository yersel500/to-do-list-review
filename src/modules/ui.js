import { taskInput } from './selectors';
import ArrayTask from './class-methods';
import printTask from './addTask';

let edit = false;
export const listofTasks = new ArrayTask();
// Empty object for task
export const objTask = {
  description: '',
  complete: '',
};

export function enableEdition(task) {
  const { description, complete, index } = task;

  // Upload content to the object
  objTask.description = description;
  objTask.complete = complete;
  objTask.index = index;

  // Fill the input
  taskInput.value = description;

  edit = true;
}

export function cleanHTML() {
  const ulContainer = document.querySelector('.list');
  while (ulContainer.firstChild) {
    ulContainer.removeChild(ulContainer.firstChild);
  }
}

export function resetObject() {
  objTask.description = '';
}

export function newTask(e) {
  const { description } = objTask;

  if (e.key === 'Enter' && description !== '') {
    if (edit) {
      listofTasks.editTask({ ...objTask });

      edit = false;
    } else {
      objTask.index = listofTasks.tasks.length + 1;
      listofTasks.addTask({ ...objTask });
    }

    listofTasks.saveLocalStorage();

    printTask(listofTasks);

    resetObject();

    taskInput.value = '';
  }
}
