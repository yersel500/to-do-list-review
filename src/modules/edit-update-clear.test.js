/**
 * @jest-environment jsdom
 */

import ArrayTask from './class-methods';
import { changeStatus } from './ui';

describe('Test editTask function', () => {
  const arraytask = new ArrayTask();
  test('Edit the object task and check if it works', () => {
    const task3 = {
      description: 'task1',
      complete: false,
      index: 0,
    };

    const task4 = {
      description: 'changed task',
      complete: false,
      index: 0,
    };

    arraytask.addTask(task3);
    arraytask.editTask(task4);
    expect(arraytask).toMatchSnapshot();
  });
});

describe('Modify the status of task', () => {
  const arraytask = new ArrayTask();
  test('The status of task should change between false/true', () => {
    const task5 = {
      description: 'task5',
      complete: false,
      index: 1,
    };

    arraytask.addTask(task5);
    changeStatus(task5);
    expect(task5.complete).toBe(true);
  });

  test('The status of task should change between false/true', () => {
    const task1 = {
      description: 'task1',
      complete: false,
      index: 1,
    };

    arraytask.addTask(task1);
    changeStatus(task1);
    changeStatus(task1);
    expect(task1.complete).toBe(false);
  });
});

describe('Clear all completed tasks', () => {
  const arraytask = new ArrayTask();
  test('All completed tasks should be removed.', () => {
    const task1 = {
      description: 'task1',
      complete: false,
      index: 1,
    };
    const task2 = {
      description: 'task2',
      complete: false,
      index: 1,
    };
    const task3 = {
      description: 'task3',
      complete: false,
      index: 1,
    };

    arraytask.addTask(task1);
    arraytask.addTask(task2);
    arraytask.addTask(task3);

    changeStatus(task1);
    changeStatus(task2);

    arraytask.deleteCompletedTask();
    expect(arraytask).toMatchSnapshot();
  });
});