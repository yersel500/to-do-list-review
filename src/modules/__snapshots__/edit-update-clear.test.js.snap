// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clear all completed tasks All completed tasks should be removed. 1`] = `
ArrayTask {
  "tasks": Array [
    Object {
      "complete": false,
      "description": "task3",
      "index": 1,
    },
  ],
}
`;

exports[`Clear all completed tasks The status of task should change between false/true 1`] = `
ArrayTask {
  "tasks": Array [
    Object {
      "complete": false,
      "description": "task3",
      "index": 1,
    },
  ],
}
`;

exports[`Test editTask function Edit the object task and check if it works 1`] = `
ArrayTask {
  "tasks": Array [
    Object {
      "complete": false,
      "description": "changed task",
      "index": 0,
    },
  ],
}
`;
